src/main/java/teammates/common/datatransfer/questions/FeedbackNumericalScaleQuestionDetails.java

```diff
@@ -488,29 +456,101 @@ public String getQuestionResultStatisticsCsv(
             if (hiddenRecipients.contains(recipient)) {
                 continue;
             }
-
-            String recipientTeam = bundle.getTeamNameForEmail(recipient);
-            boolean isRecipientGeneral = recipient.equals(Const.GENERAL_QUESTION);
-
-            Double averageScoreExcludingSelf = averageExcludingSelf.get(recipient);
-            String averageScoreExcludingSelfText =
-                    getAverageExcludingSelfText(showAvgExcludingSelf, df, averageScoreExcludingSelf);
-
-            csvBody.append(SanitizationHelper.sanitizeForCsv(recipientTeam) + ','
-                           + SanitizationHelper.sanitizeForCsv(isRecipientGeneral
-                                                      ? "General"
-                                                      : bundle.getNameForEmail(recipient))
-                           + ','
-                           + df.format(average.get(recipient)) + ','
-                           + df.format(min.get(recipient)) + ','
-                           + df.format(max.get(recipient))
-                           + (showAvgExcludingSelf ? ',' + averageScoreExcludingSelfText : "")
-                           + Const.EOL);
+            updateQuestionResultStatisticsCsv(csvBody, bundle, recipient, average,
```

Comment:
1. Does `csvBody` really need to be part of the method? Something like this should be more readable:
   ```java
   String csvEntry = ...;
   csvBody.append(csvEntry);
   ```
1. Do you really need to refactor this out? If you really do,
   - Re-consider the method parameters. A method that takes in 4 maps is itself a very good candidate for refactoring.